src/erofs/debug.rs

```diff
@@ -0,0 +1,377 @@
+use core::mem::offset_of;
+use std::{
+    collections::BTreeMap,
+    ffi::OsStr,
+    mem::discriminant,
+    os::unix::ffi::OsStrExt,
+    path::{Path, PathBuf},
+};
+
+use zerocopy::{Immutable, KnownLayout, TryFromBytes};
+
+use super::{
+    format::{self, CompactInodeHeader, ComposefsHeader, ExtendedInodeHeader, Superblock},
+    reader::{DirectoryBlock, Image, Inode, InodeHeader, InodeOps, InodeType, InodeXAttrs, XAttr},
+};
+
+macro_rules! print_fields {
+    ($ty: ty, $s: expr, $f: ident) => {{
+        let value = &$s.$f;
+        let default = if false { value } else { &Default::default() };
```

Comment:
Confused by this

Reply:
I ended up killing that code, but the idea here is that we need to coerce Default::default() to give us the correct type without using `<>` which we can't do in the macro because we have no static way to get the type.  We can't just do `!=` because this doesn't require that both sides have the same type.